### PR TITLE
Make TypeCall work with TVMScript

### PR DESCRIPTION
**Pull request by shingjan**
This PR intends to add necessary code change to synr in order to support [TVM#9492](https://github.com/apache/tvm/pull/9492)

### CODE REVIEW

**Comment by tkonolige** [comment]
@shingjan Can you add a test

**Comment by shingjan** [comment]
@tkonolige test is added.

File: synr/ast.py [hunk 1]
```diff
@@ -316,6 +316,7 @@ class TypeCall(Type):
 
     func_name: Union[Type, BuiltinOp]
     params: List[Type]
+    keyword_params: Dict[Type, Type]
```
Comment (tkonolige) [inline synr/ast.py:319]: I think this should be `keyword_params: Dict[Id, Type]` as only identifiers are allowed as keywords. (I know regular `Call` is `Dict[Expr, Expr]`, but I think it should be `Dict[Id, Expr]`.

Comment (shingjan) [reply synr/ast.py]: done

**Comment by shingjan** [comment]
@tkonolige It seems to me that if we do `Dict[Id, Type]`, this [line](                    [self._expr2type(x) for x in expr.params],) won't work while creating the keyword_params of TypeCall. There isn't a function to transform `expr` to `id`. Do you think there is any workaround on this?

**Comment by tkonolige** [comment]
Ah, sure we can change it in a later PR.

**Comment by shingjan** [comment]
@tkonolige have the change reversed. Should be good to merge after CI passes

**Comment by tkonolige** [comment]
Thanks @shingjan!

**Pull request merged**